### PR TITLE
Add support for initializer blocks

### PR DESCRIPTION
**Pull request by lazaroclapp**
* Both static and instance initializer blocks are now analyzed
  for nullability violations, same as methods.
* Static initialization is checked, and both static initializer
  blocks and @Initializer static methods are taken into account.
* Plenty of tests added.

See #28

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 1]
```diff
@@ -840,7 +887,8 @@ private FieldInitEntities collectEntities(ClassTree tree, VisitorState state) {
     Set<BlockTree> instanceInitializerBlocks = new LinkedHashSet<>();
     Set<BlockTree> staticInitializerBlocks = new LinkedHashSet<>();
     Set<MethodTree> constructors = new LinkedHashSet<>();
-    Set<MethodTree> initializerMethods = new LinkedHashSet<>();
+    Set<MethodTree> instanceInitializerMethods = new LinkedHashSet<>();
+    Set<MethodTree> staticInitializerMethods = new LinkedHashSet<>();
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java]: What is a static initializer method?  I think I intended the `staticInitializerBlocks` field to hold this info but never got around to implementing

Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java]: I am using both. staticInitializerBlocks for things like:

```
static {
    // Code
}
```

And staticInitializerMethods for:

```
@Initializer
public static init() {
    // Code
}

Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java]: Ah, ok, cool.  Maybe add a comment on what they are?

Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java]: Done :)

**Pull request merged**